lib/command.js

```diff
@@ -367,17 +383,30 @@ module.exports = function command (yargs, usage, validation, globalMiddleware) {
       const positionalKeys = Object.keys(positionalMap)
       Object.keys(positionalMap).forEach((key) => {
         [].push.apply(positionalKeys, parsed.aliases[key])
+        // currently we combine the "defaulted" object from the positional
+        // parse, from the "defaulted" object from the option parse. We may
+        // eventually merge more fields from the two parses.
+        detailedPositionalParse[key] = {
+          value: positionalMap[key],
+          defaulted: !!parsed.defaulted[key]
+        }
       })
 
       Object.keys(parsed.argv).forEach((key) => {
         if (positionalKeys.indexOf(key) !== -1) {
-          // any new aliases need to be placed in positionalMap, which
-          // is used for validation.
-          if (!positionalMap[key]) positionalMap[key] = parsed.argv[key]
+          // any new aliases need to be placed in parsedPositionals, so that
+          // the new keys pass strict validation.
+          if (!detailedPositionalParse[key]) {
+            detailedPositionalParse[key] = {
+              value: parsed.argv[key],
+              defaulted: !!parsed.defaulted[key]
```

Comment:
If detailerPositionalParse's list of fields is intended to grow, we should not have to add these fields in 2 places.

I recommand refactoring this part.